src/status.py

```diff
@@ -74,58 +79,226 @@ def write_new_status(session, status, status_level, value, national_or_sites="Na
+                font-size: 0.875rem !important;
+            }
+        </style>
+        """,
+        unsafe_allow_html=True,
+    )
+    # Fetch the latest status from the Status API
+    data_statuses = fetch_data_providers_status()
+
+    if data_statuses:
+        # Display the status of each data provider
+        for item in data_statuses:
+            provider = item["provider"]
+            source = item["source"]
+            status = item["status"]
+            msg = item["statusMessage"]
+            details = item.get("details", {})
+
+            emoji = "🟢" if status == "ok" else "🔴"
+
+            col1, col2, col3 = st.columns([3, 1, 2])
+
+            with col1:
+                st.metric(source, f"{emoji} {provider}")
+
+            with col2:
+                st.metric("Status", status)
+
+            with col3:
+                st.markdown(f"""<div style="font-size: 0.875rem">Message</div>""", unsafe_allow_html=True)
+                st.markdown(f"<div style='font-size: 0.875rem; color:#ccc; margin-top: 0.25rem;'>{msg if msg else '–'}</div>", unsafe_allow_html=True)
+
+            if provider == "EUMETSAT":
+                if details:
+                    col1, col2, col3, col4 = st.columns([3, 1, 1, 1])
+
+                    with col1:
+                        st.metric("Latest (UTC)", details.get("latestTimestamp", "-"))
+
+                    with col2:
+                        st.metric("Complete", details.get("complete", "-"))
+
+                    with col3:
+                        st.metric("Incomplete", details.get("incomplete", "-"))
+
+                    with col4:
+                        undelivered = details.get("undeliveredPlanned", 0) + details.get("undeliveredUnplanned", 0)
+                        st.metric("Undelivered", undelivered)
+                with st.expander("Show full delivery timeline"):
+                    extra = get_eumetsat_details()
+                    result_data = extra.get("details", {}).get("results", [])
+
+                    if len(result_data) >= 2:
+                        records = result_data[0]
+                        headers = result_data[1]
+                        df = pd.DataFrame(records, columns=headers)
+                        df["datetime"] = pd.to_datetime(df["datetime"])
+                        df = df.sort_values("datetime", ascending=False)
+
+                        # Add compact chart
+                        from datetime import datetime, timedelta
+                        import altair as alt
+
+                        # Melt for compact horizontal status chart
+                        df_melt = df.melt(
+                            id_vars="datetime",
+                            value_vars=["deliveryStatus", "timelyStatus"],
+                            var_name="category",
+                            value_name="status"
+                        )
+
+                        chart = alt.Chart(df_melt).mark_rect(height=12).encode(
```

Comment:
nice, I havent see altair before, but Ive been meaning to use it

Reply:
you might need to add it to the requirements.txt

Reply:
Yeah I was just looking at what was built into Streamlit already and might fit the bill for this little chart, seems nice so far!

Reply:
Yep! As in it's bundled in so you can just import it